### DISCUSSION TITLE
DataGridRow:pointerover does not work

### DISCUSSION THREAD

**Discussion by ShrutiJaiswal1494** [AvaloniaUI/Avalonia#6371] [Q&A] [answered]
Hi,
I am trying to set the background color for a Datagrid row when the mouse pointer is over them. Following the documentation, I got to know about the `:ponterover` property which is same as `Triggers` in WPF. However, this does not work for `DataGridRow`. I am using the below code. (This works for DataGridCell though). Is it not available for `DataGridRow` or is there any other way to change the background color when the pointer is over a row but that row is not selected? Any help is much appreciated. Thanks in advance.

```
<UserControl.Styles>
    <Style Selector="DataGridRow:pointerover">
      <Setter Property="Background" Value="SkyBlue"/>
    </Style>
  </UserControl.Styles>
```

**Comment by maxkatz6** [comment, marked as answer]
This selector is correct for simpler cases, but for many default controls things get more complex. See similar issue https://github.com/AvaloniaUI/Avalonia/issues/5406#issuecomment-771249842

If you will open [DataGrid / Fluent.xaml](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml#L318) theme file, you will see correct selector:
```xaml
  <Style Selector="DataGridRow:pointerover /template/ Rectangle#BackgroundRectangle">
    <Setter Property="Fill" Value="{DynamicResource SystemListLowColor}" />
  </Style>
```
See, it targets template child to modify it, instead of DataGridRow property.
I highly recommend using Avalonia styles source code as a reference, when you deal with standard controls. It simplifies life for a lot.


In details, why your code doesn't work. DataGridRow.Background value is ignored intentionally, because it represents only single state (single color value), while it needs to be used for multiple states. You can argue that you did specify ":pointerover", which would work in your case (if we didn't ignore it intentionally), but it wouldn't work if user set ```<DataGridRow Background="Blue" />``` - this row would be blue for any possible state (normal, pointer over...)

**Reply by ShrutiJaiswal1494** [reply]
This is what I was looking for. Thank you so much :)
